src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -193,10 +200,7 @@ protected override void Dispose(bool disposing)
             in ActivityCreationOptions<ActivityContext> options,
             Sampler sampler)
         {
-            var isRootSpan = /*TODO: Put back once AutoGenerateRootContextTraceId is removed.
-                              options.Parent.TraceId == default ||*/ options.Parent.SpanId == default;
-
-            if (sampler != null)
+            if (sampler != null && !(sampler is AlwaysOffSampler))

```

Comment:
Could we also special case AlwaysOffSampler? I think we don't need to go through all the logic here.

Reply:
In general, I think all the built-in samplers can be special cased. Given sampler is running on the hot path, it will worth every single instruction level micro-optimization.

Reply:
He's got a special case in there for the `AlwaysOffSampler` 😄

Reply:
I mean if we can special case `AlwaysOffSampler` to save two function calls (which translates to better CPU cache hit), it might be a good deal.